command/agent/auth/jwt/jwt.go

```diff
@@ -97,6 +98,19 @@ func NewJWTAuthMethod(conf *auth.AuthConfig) (auth.AuthMethod, error) {
 	if j.removeJWTAfterReading {
 		readPeriod = 500 * time.Millisecond
 	}
+
+	// This is test-only config, so that tests don't need to wait for minutes
+	// in the case that removeJWTAfterReading != true
+	if testOverrideReadPeriodRaw, ok := conf.Config["test_override_read_period"]; ok {
```

Comment:
I like this idea - though I'm not sure if it would help much in production, it certainly can't hurt.

It seems like a great way to avoid 'prod code cares about tests' while giving an additional bit of config that could help users in the future. Thanks!

Reply:
Done! Thanks for the suggestion.